packages/beacon-node/src/network/processor/gossipHandlers.ts

```diff
@@ -413,8 +414,12 @@ export async function validateGossipFnRetryUnknownRoot<T>(
         e instanceof AttestationError &&
         e.type.code === AttestationErrorCode.UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT
       ) {
-        if (unknownBlockRootRetries++ < MAX_UNKNOWN_BLOCK_ROOT_RETRIES) {
+        if (unknownBlockRootRetries === 0) {
           // Trigger unknown block root search here
+          const rootHex = toHexString(blockRoot);
+          events.emit(NetworkEvent.unknownBlock, {rootHex});
```

Comment:
How does this safeguard against spamming unknownBlock events if multiple attestations reference to the same unknown block root

Reply:
@dapplion I addressed this in the latest commit, there is a cache inside NetworkProcessor to only search for 1 root per slot, and `NetworkEvent.unknownBlock` is triggered in a single place